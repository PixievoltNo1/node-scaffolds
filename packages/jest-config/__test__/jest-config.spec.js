const path = require('path')
const { resolveModuleNameMapper, tsMonorepoConfig } = require('..')

const originalCwd = process.cwd()
afterEach(() => {
  process.chdir(originalCwd)
})

const resolveFixturePath = p => path.join(__dirname, 'fixtures', p)
const desensitizeModuleNameMapper = moduleNameMapper => {
  const result = {}
  for (const key of Object.keys(moduleNameMapper)) {
    if (Array.isArray(moduleNameMapper[key])) {
      result[key] = moduleNameMapper[key].map(
        p => '<WORKSPACE>/' + path.relative(__dirname, p),
      )
    } else {
      result[key] =
        '<WORKSPACE>/' + path.relative(__dirname, moduleNameMapper[key])
    }
  }
  return result
}

describe('resolveModuleNameMapper', function () {
  test('basic', function () {
    expect(
      desensitizeModuleNameMapper(
        resolveModuleNameMapper(resolveFixturePath('basic')),
      ),
    ).toMatchSnapshot('basic')
  })

  test('custom tsconfig name', function () {
    expect(
      desensitizeModuleNameMapper(
        resolveModuleNameMapper(
          resolveFixturePath('custom-tsconfig-name'),
          'tsconfig.src.json',
        ),
      ),
    ).toMatchSnapshot('custom-tsconfig-name')
  })

  test('empty', function () {
    expect(resolveModuleNameMapper(resolveFixturePath('empty'))).toEqual({})
  })

  test('not exists', function () {
    expect(
      resolveModuleNameMapper(resolveFixturePath('empty'), 'tsconfig.json'),
    ).toEqual({})
  })
})

test('tsMonorepoConfig', function () {
  const rootDir = resolveFixturePath('basic')
  process.chdir(rootDir)
  const config = tsMonorepoConfig(rootDir)
  config.moduleNameMapper = desensitizeModuleNameMapper(config.moduleNameMapper)
  expect(config).toMatchSnapshot()
})
