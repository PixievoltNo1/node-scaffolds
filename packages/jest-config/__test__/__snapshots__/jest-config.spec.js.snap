// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolveModuleNameMapper basic: basic 1`] = `
{
  "^@/(.+)$": "<WORKSPACE>/fixtures/basic/src/$1",
}
`;

exports[`resolveModuleNameMapper custom tsconfig name: custom-tsconfig-name 1`] = `
{
  "^@/(.+)$": [
    "<WORKSPACE>/fixtures/custom-tsconfig-name/src/$1",
    "<WORKSPACE>/fixtures/custom-tsconfig-name/script/$1",
  ],
}
`;

exports[`tsMonorepoConfig 1`] = `
{
  "bail": true,
  "collectCoverage": false,
  "collectCoverageFrom": [
    "<rootDir>/cli.js",
    "<rootDir>/index.js",
    "<rootDir>/src/*.{js,jsx,ts,tsx}",
    "<rootDir>/src/**/*.{js,jsx,ts,tsx}",
  ],
  "coverageDirectory": "<rootDir>/coverage/",
  "coveragePathIgnorePatterns": [],
  "coverageProvider": "v8",
  "coverageReporters": [
    "text",
    "text-summary",
  ],
  "coverageThreshold": {
    "global": {
      "branches": 80,
      "functions": 80,
      "lines": 80,
      "statements": 80,
    },
  },
  "errorOnDeprecated": true,
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js",
    "jsx",
    "json",
    "node",
  ],
  "moduleNameMapper": {
    "^@/(.+)$": "<WORKSPACE>/fixtures/basic/src/$1",
  },
  "roots": [
    "<rootDir>/src",
  ],
  "testEnvironment": "node",
  "testEnvironmentOptions": {
    "url": "http://localhost/",
  },
  "testPathIgnorePatterns": [
    "/coverage/",
    "/lib/",
    "/node_modules/",
  ],
  "testRegex": "/(__test__)/[^/]+\\.spec\\.[jt]sx?$",
  "transform": {
    "^.+\\.tsx?$": [
      "ts-jest",
      {
        "tsconfig": "<rootDir>/tsconfig.json",
        "useESM": undefined,
      },
    ],
  },
  "verbose": true,
}
`;
